test/unit/test_quil.py

```diff
@@ -1306,3 +1310,30 @@ def test_params_pi_and_precedence():
     prog = Program(f"RX({more_less_trivial_pi}) 0")
     exp = str(prog[0].params[0])
     assert _eval_as_np_pi(more_less_trivial_pi) == _eval_as_np_pi(exp)
+
+
+class TestProgram:
+    def test_calibrations(self, snapshot: SnapshotAssertion):
+        program = Program(
+            "DEFCAL Calibrate 0:\n\tX 0",
+            DefCalibration("Reticulating-Splines", [Parameter("Spline")], [Qubit(1)], [Y(1)]),
+            DefMeasureCalibration(Qubit(2), MemoryReference("theta"), [Z(2)]),
+        )
+
+        calibrations = program.calibrations
+        measure_calibrations = program.measure_calibrations
+        assert all((isinstance(cal, DefCalibration) for cal in program.calibrations))
+        assert all((isinstance(cal, DefMeasureCalibration) for cal in program.measure_calibrations))
+        assert calibrations == snapshot
+        assert measure_calibrations == snapshot
+
+    def test_frames(self, snapshot: SnapshotAssertion):
+        program = Program(
+            'DEFFRAME 1 "frame":\n\tcenter_frequency: 440',
```

Comment:
FWIW, in practice, this is `CENTER-FREQUENCY` (`-` not `_`). Would be good to make sure nothing coerces that to a `_`, clobbering the value you've set here

Reply:
Ah, good catch. My re-implementation of `DefFrame` was using the lower snake case variants too. Fixed them up all around.